### DISCUSSION TITLE
SmartLockPro3 - switch from native internal MQTT to Nuki Hub hybrid mode

### DISCUSSION THREAD

**Discussion by Ra72xx** [technyon/nuki_hub#437] [General]
Subject says it all: I have my SmartLockPro3 already paired via MQTT to Home Assistant  and now I'd like to switch to the "hybrid mode".  I followed the instructions in [https://github.com/technyon/nuki_hub/blob/master/HYBRID.md](url), however I did not follow the point `Make sure NOT to enable Auto discovery, Home Assistant will be setup using Nuki Hub` , as the the smartlock was at this point already auto-discovered by Home Assistant long ago. 
Nevertheless, everything seems to work, I now have two smartlock entities, one for direct control via MQTT and one controlled by Nuki Hub (as expected). Both work and seem fine to me.
I'm just wondering, as I see no special error outputs etc. by Nuki Hub: Am I missing some hybrid mode features by this way of setup? E.g. is Nuki Hub still able to control the lock both by MQTT and by BLE, or is MQTT missing? I see no information in the UI to find this out.

**Comment by iranl** [comment]
As the hybrid instructions tell you if you want to make sure hybrid mode is setup correctly:

> Test that state changes are recieved and processed by Nuki Hub by looking at the MQTT topics using an application like MQTT Explorer

Check that changes on the official mqtt nuki topics are mirrored almost immediately on the nuki hub mqtt topics. And that when you change the lock state to open or closed in home assistant using the nuki hub device that this changes the lockaction value of both the nuki hub and official mqtt device topics.

You can also enable mqtt logging in the nuki hub config and look for mqtt messages on the nukihub/maintenance/log mqtt topic about receiving updates from the official implementation.

Hybrid mode only adds speed in picking up (intermediate) state changes of the lock and executing lock actions through Nuki Hub. All information and functionality that is not available through the official mqtt implementation (e.g. config, auth, keypad, timecontrol info and changes) will still be done over bluetooth.

The downside of your setup is that it can cause confusion which lock device to use.
You also can't do ACL on the official mqtt implementation (e.g. disabling locking/unlocking/unlatching from mqtt/ha), which you can do using Nuki Hub. If you disable unlatching in Nuki Hub this will mot be executed on the lock when using the nuki hub device, but will be done if you use the official nuki device in ha

**Comment by Ra72xx** [comment]
Thanks for the explanation. So basically "hybrid mode" uses official MQTT whenever possible and BLE when MQTT is not available and/or the functionality demands it. But if I understand that correctly, the Nuki Hub MQTT simply shows all events and it is not really discernible if the used transport was BLE or official MQTT.

I'm just a bit unsure because of the line "disable autodiscovery" in the docs. Where would I disable autodiscovery of the official Nuki MQTT at all, and why should I? Is this ony a cosmetic problem (two locks for one device) or is it a functional problem?

**Reply by iranl** [reply]
The last paragraph of my answer explains why you should disable auto discovery for the official Nuki implementation. 

It will work when both are enabled, but depending on what functions you want Nuki Hub to perform can prevent some functionality from being properly enforced (such as limiting what lock actions can be performed). If you don't use these functions there is no real harm in having auto discovery for both enables.

How to disable auto discovery for the official Nuki implementation is explained on https://support.nuki.io/hc/en-us/articles/14052016143249-Activate-MQTT-via-the-Nuki-app

**Reply by Ra72xx** [reply]
Ah, OK, I guess these official instructions have changed since I set up my Nuki Lock (there were less options at that time, e.g. no MQTT over thread etc. ...).
